App.Tests/TestWeddingService.cs

```diff
@@ -0,0 +1,180 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using App.Models;
+using App.Services;
+
+namespace App.Tests
+{
+    public class TestWeddingService
+    {
+        [Fact]
+        public void testDayIndexWhenEveryoneCanCome_SundayAndSaturdayAreFree()
+        {
+            List<Guest> guests = new List<Guest>();
+            guests.Add(new Guest("Inga", new List<int>{1, 3}));
+            guests.Add(new Guest("Sasha", new List<int>{1, 2, 3, 4, 5}));
+
+            WeddingService weddingService = new WeddingService();
+            int dayIndex = weddingService.dayIndexWhenEveryoneCanCome(guests);
+            Assert.Equal(7, dayIndex);
+        }
+
+        [Fact]
+        public void testDayIndexWhenEveryoneCanCome_onlyMondayIsFree()
+        {
+            List<Guest> guests = new List<Guest>();
+            guests.Add(new Guest("Inga", new List<int>{3, 7, 6}));
+            guests.Add(new Guest("Sasha", new List<int>{2, 3, 4, 5}));
+
+            WeddingService weddingService = new WeddingService();
+            int dayIndex = weddingService.dayIndexWhenEveryoneCanCome(guests);
+            Assert.Equal(1, dayIndex);
+        }
+
+        [Fact]
+        public void testDayIndexWhenEveryoneCanCome_allBusy()
+        {
+            List<Guest> guests = new List<Guest>();
+            guests.Add(new Guest("Inga", new List<int>{1, 6, 7}));
+            guests.Add(new Guest("Sasha", new List<int>{2, 3, 4, 5}));
+
+            WeddingService weddingService = new WeddingService();
+            int dayIndex = weddingService.dayIndexWhenEveryoneCanCome(guests);
+            Assert.Equal(0, dayIndex);
+        }
+
+        [Fact]
+        public void testDayIndexWhenEveryoneCanCome_AllBusyEveryday()
+        {
+            List<Guest> guests = new List<Guest>();
+
+            guests.Add(new Guest("Inga", new List<int>{1, 2, 3, 4, 5, 6, 7}));
+            guests.Add(new Guest("Sasha", new List<int>{1, 2, 3, 4, 5, 6, 7}));
+
+            WeddingService weddingService = new WeddingService();
+            int dayIndex = weddingService.dayIndexWhenEveryoneCanCome(guests);
+            Assert.Equal(0, dayIndex);
+        }
+
+        [Fact]
+        public void testDayIndexWhenEveryoneCanCome_allFree()
+        {
+            List<Guest> guests = new List<Guest>();
+            guests.Add(new Guest("Inga", new List<int>{}));
+            guests.Add(new Guest("Sasha", new List<int>{}));
+
+            WeddingService weddingService = new WeddingService();
+            int dayIndex = weddingService.dayIndexWhenEveryoneCanCome(guests);
+            Assert.Equal(7, dayIndex);
+        }
+
+        [Fact]
+        public void testDayIndexWhenEveryoneCanCome_noGuests()
+        {
+            List<Guest> guests = new List<Guest>();
+
+            WeddingService weddingService = new WeddingService();
+            int dayIndex = weddingService.dayIndexWhenEveryoneCanCome(guests);
+            Assert.Equal(7, dayIndex);
+        }

```

Comment:
Tests for `dayIndexWhenEveryoneCanCome` method look very similar and most of the code is duplicated. Consider using Parameterised tests (called Theories in XUnit), [MemberData](https://hamidmosalla.com/2017/02/25/xunit-theory-working-with-inlinedata-memberdata-classdata/) might be a good candidate here